contrib/opencensus-context/opencensus/common/runtime_context/__init__.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import threading
+
+__all__ = ['RuntimeContext']
+
+class _RuntimeContext(object):
+    @classmethod
+    def clear(cls):
+        raise NotImplementedError  # pragma: NO COVER
+
+    @classmethod
+    def register_slot(cls, name, default = None):
+        raise NotImplementedError  # pragma: NO COVER
+
+    def __init__(self):
+        pass
+
+    def __repr__(self):
+        return ('{}({})'
+                .format(
+                    type(self).__name__,
+                    dict((n, self._slots[n].get()) for n in self._slots)
+                ))
+
+    def __getattr__(self, name):
+        if name not in self._slots:
+            raise AttributeError('{} is not a registered context slot'.format(name))
+        slot = self._slots[name]
+        return slot.get()
+
+    def __setattr__(self, name, value):
+        if name not in self._slots:
+            raise AttributeError('{} is not a registered context slot'.format(name))
+        slot = self._slots[name]
+        slot.set(value)
+
+class _TlsRuntimeContext(_RuntimeContext):
```

Comment:
What is TLS?

Reply:
Thread-local Storage https://en.wikipedia.org/wiki/Thread-local_storage. Probably need a better name here.

Reply:
Changed to ThreadLocal.